src/Service/MemoryInfoHandler.cpp

```diff
@@ -10,20 +10,66 @@
 namespace orbit_service {
 
 void MemoryInfoHandler::Start(orbit_grpc_protos::CaptureOptions capture_options) {
-  CHECK(memory_info_producer_ == nullptr);
-
   if (!capture_options.collect_memory_info()) return;
 
-  memory_info_producer_ = std::make_unique<orbit_memory_tracing::MemoryInfoProducer>(
-      capture_options.memory_sampling_period_ns());
-  memory_info_producer_->SetListener(this);
-  memory_info_producer_->Start();
+  CHECK(system_memory_info_producer_ == nullptr);
+  system_memory_info_producer_ = std::make_unique<orbit_memory_tracing::MemoryInfoProducer>(
+      capture_options.memory_sampling_period_ns(), capture_options.pid(),
+      orbit_memory_tracing::SystemMemoryInfoProducerRunFn);
+  system_memory_info_producer_->SetListener(this);
+  system_memory_info_producer_->SetThreadName("SysMemPr::Run");
```

Comment:
I think this can be extracted to a factory method, something like:
```
system_memory_info_producer_ = orbit_memory_tracing::CreateSystemMemoryInfoProducer(this, capture_options.memory_sampling_period_ns(), capture_options.pid());
system_memory_info_producer_ ->Start();
```

And have it plug other things in. same for other memory producers.

You can define factory methods next to MemoryInfoProducer class definition.

Reply:
Good idea! Changed accordingly. Thanks!